### PR TITLE
Add new gulp task to 'gulpfile.js' for running mongo daemon.

### PR DESCRIPTION
**Pull request by llighter**
I applied the mongo daemon startup option to 'gulpfile.ts'
ISSUE= #503

### CODE REVIEW

File: gulpfile.ts [hunk 1]
```diff
@@ -75,3 +76,17 @@ gulp.task('build_server', () => {
     .js
     .pipe(gulp.dest('out/'));
 });
+
+const mongodb_path = './third_party/mongodb';
+gulp.task('start-mongo', runCommand(`${mongodb_path}/bin/mongod --dbpath ${mongodb_path}`));
```
Comment (romandev) [inline gulpfile.ts]: We are already using gulp.shell. So, can we just use gulp.shell() instead of runCommand()?

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "@types/request": "^2.0.3",
     "@types/run-sequence": "0.0.29",
     "@types/supertest": "^2.0.3",
+    "child_process": "^1.0.2",
```
Comment (romandev) [inline package.json]: Even If we have to child_processs, do we really need this? The child_process modules is built-in library as far as I know. WDYT?

File: gulpfile.ts (hunk 1 above)
Comment (llighter) [reply gulpfile.ts]: Oh, I didn't know about 'gulp.shell'. I will look into it and fix it.

File: package.json (hunk 2 above)
Comment (llighter) [reply package.json]: If we can use the built-in library, it's a good idea to use it. I'll look into it too. :)

File: gulpfile.ts (hunk 1 above)
Comment (romandev) [reply gulpfile.ts]: @llighter I had to give you the information early but I couldn't. I'm sorry for that. Also, thank you for this work!

Comment (llighter) [reply gulpfile.ts]: @romandev It's fine. I was able to study this in more detail.

File: package.json (hunk 2 above)
Comment (romandev) [reply package.json]: But if you use gulp.shell, we might not need it :)

File: gulpfile.ts (hunk 1 above)
Comment (llighter) [reply gulpfile.ts]: @romandev I've found that now gulp do not recommend 'gulp.shell' anymore.
Here is [blackList.json file]( https://github.com/gulpjs/plugins/blob/master/src/blackList.json) at gulpjs repo.
Gulp recommend to use [gulp-exec]() or [child_process]().
And according to [gulp-exec]() docs, If you just want to run a command, don't use this plugin.
And [gulp API docs](https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpdestpath-options) use [child_process](https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpdestpath-options) like this.
```
// run a command in a shell
var exec = require('child_process').exec;
gulp.task('jekyll', function(cb) {
  // build Jekyll
  exec('jekyll build', function(err) {
    if (err) return cb(err); // return error
    cb(); // finished task
  });
});
```
What should I use?
WDYT? PTAL :)

Comment (romandev) [reply gulpfile.ts]: Okay! Thank you for the information. If so, we can move this patch to gulp-exec or child_process.
And, once this patch is merged, you can also replace gulp-shell with gulp-exec or child_process in existing codes.

Comment (llighter) [reply gulpfile.ts]: @romandev ok, then I will replace gulp-shell with child_process in our code.

**Comment by llighter** [comment]
I'll close this PR for new PR.

**Pull request closed without merging**